event_track_info/models/product_training_plan.py

```diff
@@ -8,8 +8,47 @@ class ProductTrainingPlan(models.Model):
     _inherit = 'product.training.plan'
 
     def _search_product_training_plan(self, product, num_session):
-        training = self.search([
-            '|', '&', ('product_tmpl_id', '=', product.product_tmpl_id.id),
-            ('product_id', '=', False), ('product_id', '=', product.id),
-            ('sequence', '=', num_session)], limit=1)
-        return training
+        data = {}
+        cond = [('product_tmpl_id', '=', product.product_tmpl_id.id),
+                ('product_id', '=', False),
+                ('sequence', '=', num_session)]
+        data = self._catch_training_plan_information(
+            data, self.search(cond))
+        cond = [('product_tmpl_id', '=', product.product_tmpl_id.id),
+                ('product_id', '=', product.id),
+                ('sequence', '=', num_session)]
+        data = self._catch_training_plan_information(
+            data, self.search(cond))
+        return data
+
+    def _catch_training_plan_information(self, data, trainings):
+        for training in trainings:
+            if data.get('name', False):
+                if training.training_plan_id.name not in data.get('name'):
+                    data['name'] = u'{}\n\n{}'.format(
+                        data.get('name'), training.training_plan_id.name)
+            else:
+                data['name'] = training.training_plan_id.name
+            if data.get('url', False):
+                data['url'] = u'{}\n\n{}'.format(
+                    data.get('url'), training.training_plan_id.url)
+            else:
+                data['url'] = training.training_plan_id.url
```

Comment:
`data['url'] = (data.get('url', False) and data['url'] = u'{}\n\n{}'.format(data.get('url'), training.training_plan_id.url) or training.training_plan_id.url)`